### DISCUSSION TITLE
v1.0.0-RC4

### DISCUSSION THREAD

**Discussion by jatcwang** [tpolecat/doobie#1886] [General]
Please post any upgrade questions for v1.0.0-RC4 here :)

**Comment by jatcwang** [comment]
If you're using PostgreSQL and you're seeing errors like these

```
✕ C08 created_at                 TIMESTAMPWITHTIMEZONE (timestamptz)                                   NOT NULL  →  Timestamp
    TIMESTAMPWITHTIMEZONE (timestamptz) is not coercible to Timestamp
    according to the JDBC specification or any defined mapping. Fix
    this by changing the schema type to TIMESTAMP; or the Scala type
    to an appropriate object type.
```

It's probably because you're using `doobie.util.meta.LegacyInstantMetaInstance` somewhere to provide `Meta[Instant]` and it is complaining because this legacy instance uses `java.sql.Timestamp` under the hood. `java.sql.Timestamp` is [ill-specified](https://stackoverflow.com/a/63078938/2036669) and should be avoided.

You should switch to use `Meta[Instant]` in `doobie.postgres.JavaTimeInstances` instead, where assumes driver-native support for Java 8 types.

**Comment by krrrr38** [comment]
@jatcwang 
First of all, thank you for maintain doobie. This is one of the nice library to handle sql correctly
If possible, please release v1.0.0. Some users including me want it to upgrade cats. Thanks.

**Comment by jatcwang** [comment]
Thanks for the nice words @krrrr38. We're slowing working towards it which you can track in
https://github.com/tpolecat/doobie/milestones

I don't think there's anything stopping you from upgrading cats / cats-effect versions though? You should just be able to specify the latest version of cats/cats-effect them in your build and it'll all work

**Reply by krrrr38** [reply]
Thank you for you reply. I will check remaining tasks on 1.0.

> specify the latest version

Sure. If doobie 0.13 has compatible with cats 3.x, I will try it. Thanks!

**Reply by jatcwang** [reply]
Ah you're on 0.13.x. Yeah in that case you can either upgrade to the latest 1.0 RC version + cats-effect 3, or wait for doobie 1.0.0 and do the upgrade then. It's worth considering doing the former now so you're not depending on cats-effect 2, because many other libraries is using cats-effect 3

**Reply by krrrr38** [reply]
I see. In my situation, we can test RC version on non-production env, but generally it is difficult to use it on production env. Thank you for letting me know the information.

**Reply by Daenyth** [reply]
@krrrr38 I would say that doobie's RC with cats-effect 3 is actually *more* appropriate for production than the 0.13.x line, especially when combined with the overwhelmingly massive number of improvements across the ecosystem. Cats-effect 2 has been EOL for multiple years at this point. The "RC" is just a label, and if your bosses need convincing you can probably show download stats from repository mirrors that show the RC versions are used more than the old stuff

**Reply by misja-alma** [reply]
If the RC is just a label, then why not release 1.0?
I'm a bit concerned about the amount of RC versions that it takes to get to 1.0, especially because it seems new features are introduced between RC updates. That's not what RC's are meant for, they are meant for ironing out the last bugs in the final version.
The problem is that some companies simply have as policy only to release final versions in Prod. If that makes sense is an interesting discussion, but this is just the way things are in the corporate world.

**Reply by jatcwang** [reply]
1.0 isn't a label it's a commitment to binary compatibility too, which I don't think we were in a good state to guaranatee.
I am sympathetic though regarding "no RC policy", and in hindsight maybe I should've released an imperfect 1.0 and put the rest in 2.0. It will just will be a very frequent 2.0, 3.0... etc given how many important changes are being contributed.
